sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercionSuite.scala

```diff
@@ -1244,48 +1244,6 @@ class TypeCoercionSuite extends AnalysisTest {
     )
   }
 
-  test("BooleanEquality simplification") {
-    val be = TypeCoercion.BooleanEquality
```

Comment:
Are we able to keep the same tests?

Reply:
Since we consolidated the simplification into UnwrapCastInBinaryComparison, we have equivalent tests added in UnwrapCastInBinaryComparisonSuite.scala 
https://github.com/apache/spark/pull/33865/files#diff-cecb9a54bc9379bfe46610ca54215ef5921b512ca50c8630ec6dbb4b8576fdd1R306

Reply:
For the rest of BooleanEquality functionalities we kept, the tests are still available at
https://github.com/apache/spark/pull/33865/files#diff-e4126ea0e90478e1a6a71b120e71d720463f31bf33fb1ef943989c2f37294ebaL1221